jerry-core/ecma/builtin-objects/ecma-builtin-helpers.c

```diff
@@ -667,100 +770,47 @@ ecma_builtin_helper_string_prototype_object_index_of (ecma_string_t *original_st
 bool
```

Comment:
I suggest to use the return value as uint32_t with the matching index and `UIN32T_MAX` for not matching.